### ISSUE TITLE
Building fails on Cygwin

### ISSUE THREAD

**Issue by neheb** [sayurin/optipng-zopfli#1]
Log:

$ make
cd src/optipng && \
make && \
cd ../..
make[1]: Entering directory `/home/mangix/devstuff/optipng-zopfli/src/optipng'
gcc -c  -O2 -Wall -Wextra -I../cexcept  -I../zlib  -I../libpng  -I../opngreduc  -I../pngxtern -o optipng.o optipng.c
In file included from optipng.c:33:0:
../libpng/png.h:423:28: fatal error: zlib_container.h: No such file or directory
compilation terminated.
Makefile:100: recipe for target`optipng.o' failed
make[1]: **\* [optipng.o] Error 1
make[1]: Leaving directory `/home/mangix/devstuff/optipng-zopfli/src/optipng'
Makefile:2: recipe for target`all' failed
make: **\* [all] Error 2

**Comment by neheb** [comment]
After attempting to fix it by placing all of the files in the zopfli directory into libpng, I now get this:

$ make
cd src/optipng && \
make && \
cd ../..
make[1]: Entering directory `/home/mangix/devstuff/optipng-zopfli/src/optipng'
gcc -s -o optipng optipng.o  optim.o  cbitset.o  osys.o  wildargs.o ../opngreduc/libopngreduc.a  ../pngxtern/libpngxtern.a  ../libpng/libpng.a  ../zlib/libz.a  ../gifread/libgifread.a  ../pnmio/libpnmio.a  ../minitiff/libminitiff.a   -lm
../libpng/libpng.a(pngwrite.o):pngwrite.c:(.text+0xe45): undefined reference to`ZlibCompress'
../libpng/libpng.a(pngwutil.o):pngwutil.c:(.text+0x76): undefined reference to `ZlibCompress'
../libpng/libpng.a(pngwutil.o):pngwutil.c:(.text+0x700): undefined reference to`InitOptions'
../libpng/libpng.a(pngwutil.o):pngwutil.c:(.text+0x2814): undefined reference to `ZlibCompress'
/usr/lib/gcc/i686-pc-cygwin/4.7.2/../../../../i686-pc-cygwin/bin/ld: ../libpng/libpng.a(pngwutil.o): bad reloc address 0x168 in section`.rdata'
/usr/lib/gcc/i686-pc-cygwin/4.7.2/../../../../i686-pc-cygwin/bin/ld: final link failed: Invalid operation
collect2: error: ld returned 1 exit status
Makefile:97: recipe for target `optipng' failed
make[1]: *** [optipng] Error 1
make[1]: Leaving directory`/home/mangix/devstuff/optipng-zopfli/src/optipng'
Makefile:2: recipe for target `all' failed
make: **\* [all] Error 2

**Comment by gjedeer** [comment]
This is by no means an automatic build. The missing symbols are from libzopfli.a, which is not built by the makefiles. The makefile in src/zopfli errors out too (it's for MSVC). This is the makefile I ended up with:

https://gist.github.com/gjedeer/7080715

Then I had to modify the optipng makefile to include the library produced. I'm not commiting it back because it's all a giant hack.

**Comment by lorenzhs** [comment]
Yes, it would be nice to have a cross-platform build process. I ended up doing the same thing as @gjedeer to build it on Debian, and it really is a giant hack.
